internal/entities/properties/service/service.go

```diff
@@ -337,6 +333,75 @@ func (ps *propertiesService) EntityWithProperties(
 	return models.NewEntityWithProperties(ent, props), nil
 }
 
+func (ps *propertiesService) EntityWithPropertiesByID(
+	ctx context.Context, entityID uuid.UUID,
+	opts *CallOptions,
+) (*models.EntityWithProperties, error) {
+	q := ps.getStoreOrTransaction(opts)
+
+	zerolog.Ctx(ctx).Debug().Str("entityID", entityID.String()).Msg("fetching entity with properties")
+	ent, err := q.GetEntityByID(ctx, entityID)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, ErrEntityNotFound
+	} else if err != nil {
+		return nil, fmt.Errorf("error getting entity: %w", err)
+	}
+
+	return ps.getEntityWithProperties(ctx, ent, opts)
+}
+
+func (ps *propertiesService) EntityWithPropertiesByUpstreamID(
+	ctx context.Context, entType minderv1.Entity,
+	upstreamID string, hint ByUpstreamIdHint,
+	opts *CallOptions,
```

Comment:
I'm a little confused by this function. Don't we want to use properties always instead of having to pass the upstream ID? The provider webhook handler would parse al relevant properties that are necessary to identify an entity, and that would be used here.

Reply:
This is what I started with actually but then always ended up using only upstreamID as the attribute. This might just be a function of us using the method in the github provider where we typically look up entities by ID.

I'm fine adding properties back and handling only by-ID for now or by-ID and by-name to look up the entity in the cache.

Looking at the code again we could also just reuse getEntityIdByProperties and in the existing callers (RetrieveAllProperties and RetrieveProperty) just set the hint to the projectID and providerID which would point them at a single entity only.

Thanks for the feedback!

Reply:
Done, now we just loop over the get-by-something properties in a defined order and try to look them up.